File: Data Structures & Algorithms/anagram-groups/submission-7.py
from collections import defaultdict
class Solution:

    def getMapping(self, string: str) -> str:
        return ''.join(sorted(string))

    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        group: dict[dict, list[str]] = defaultdict(list)
        for string in strs:
            # print(self.getMapping(string))
            group[self.getMapping(string)].append(string)
        
        return [grouping for grouping in group.values()]

File: Data Structures & Algorithms/is-anagram/submission-7.cpp
#include <array>
class Solution {
public:
    bool isAnagram(string s, string t) {
        array<int, 26> sV = {0};
        array<int, 26> tV = {0};

        for(const auto& c : s){
            sV[c - 'a']++;
        }

        for(const auto& c : t){
            tV[c - 'a']++;
        }

        return sV == tV;

    }
};
